File: 1510-find-lucky-integer-in-an-array/1510-find-lucky-integer-in-an-array.java
class Solution {
    public int findLucky(int[] arr) {
        Map<Integer, Integer> m = new TreeMap<>(Collections.reverseOrder());

        for (int a : arr) {
            m.put(a, m.getOrDefault(a, 0) + 1);
        }

        for (Map.Entry<Integer, Integer> entry : m.entrySet()) {
            int key = entry.getKey();
            int val = entry.getValue();

            if(key == val) {
                return key;
            }
        }

        return -1;
    }
}

File: 1510-find-lucky-integer-in-an-array/1510-find-lucky-integer-in-an-array.ts
function findLucky(arr: number[]): number {
    arr.sort((a, b) => a - b)

    let freq = 1;

    for (let i = arr.length - 1; i >= 0; i--) {
        freq = 1;
        const current = arr[i];
        if(current==1) {

        }
        while (arr[i] == arr[i - 1]) {
            freq++
            i--;
        }
        if (freq == current) {
            return current;
        }
    }

    return -1;
};